src/main/java/com/exactpro/th2/common/event/Event.java

```diff
@@ -302,16 +343,93 @@ protected String formatStateException(String fieldName, Object value) {
         return fieldName + " in event '" + id + "' already sed with value '" + value + '\'';
     }
 
-    @NotNull
+    /**
+     * @deprecated use {@link #getAggregatedStatus} instead
+     */
+    @Deprecated(forRemoval = true)
     protected Status getAggrigatedStatus() {
+        return getAggregatedStatus();
+    }
+
+    @NotNull
+    protected Status getAggregatedStatus() {
         if (status == Status.PASSED) {
-            return subEvents.stream().anyMatch(subEvent -> subEvent.getAggrigatedStatus() == Status.FAILED)
+            return subEvents.stream().anyMatch(subEvent -> subEvent.getAggregatedStatus() == Status.FAILED)
                     ? Status.FAILED
                     : Status.PASSED;
         }
         return Status.FAILED;
     }
 
+    private void batch(int maxEventBatchContentSize, List<EventBatch> result, Map<EventID, List<com.exactpro.th2.common.grpc.Event>> eventGroups, @Nullable EventID eventID) throws JsonProcessingException {
+        eventID = ObjectUtils.defaultIfNull(eventID, DEFAULT_EVENT_ID);
+        EventBatch.Builder builder = setParentId(EventBatch.newBuilder(), eventID);
+
+        List<com.exactpro.th2.common.grpc.Event> events = Objects.requireNonNull(eventGroups.get(eventID),
+                eventID != DEFAULT_EVENT_ID
+                        ? "Neither of events refers to " + shortDebugString(eventID)
+                        : "Neither of events is root event");
+
+        for (var protoEvent : events) {
+            var checkedProtoEvent = checkAndRebuild(maxEventBatchContentSize, protoEvent);
+
+            LOGGER.trace("Process {} {}", checkedProtoEvent.getName(), checkedProtoEvent.getType());
+            if(eventGroups.containsKey(checkedProtoEvent.getId())) {
```

Comment:
I wonder what will happen if the event won't have an ID set. I think we will get infinite recursion. But that should be a problem I hope

Reply:
Added test